File: VB/WebDocumentViewer_UserName/Services/CustomPageInfoDataProvider.vb
Imports System.Web
Imports DevExpress.XtraPrinting

Namespace WebDocumentViewer_UserName.Services
    Friend Class CustomPageInfoDataProvider
        Inherits PageInfoDataProviderBase

        Private ReadOnly httpContext As HttpContext
        Public Sub New(ByVal httpContext As HttpContext)
            Me.httpContext = httpContext
        End Sub

        Public Overrides Function GetText(ByVal ps As PrintingSystemBase, ByVal brick As PageInfoTextBrickBase) As String
            If brick.PageInfo <> PageInfo.UserName Then
                Return Nothing
            End If
            If httpContext Is Nothing Then
                Return "<No Information>"
            End If
            Dim user = httpContext.User
            If user Is Nothing OrElse user.Identity Is Nothing Then
                Return "<Please enable Forms or Windows security>"
            End If
            Dim identity = user.Identity
            Return If(identity.IsAuthenticated, identity.Name, "<Guest>")
        End Function
    End Class
End Namespace

File: CS/WebDocumentViewer_UserName/Services/CustomPageInfoDataProvider.cs
using System.Web;
using DevExpress.XtraPrinting;

namespace WebDocumentViewer_UserName.Services {
    class CustomPageInfoDataProvider : PageInfoDataProviderBase
    {
        readonly HttpContext httpContext;
        public CustomPageInfoDataProvider(HttpContext httpContext)
        {
            this.httpContext = httpContext;
        }

        public override string GetText(PrintingSystemBase ps, PageInfoTextBrickBase brick)
        {
            if(brick.PageInfo != PageInfo.UserName) {
                return null;
            }
            if(httpContext == null)
                return "<No Information>";
            var user = httpContext.User;
            if(user == null || user.Identity == null)
                return "<Please enable Forms or Windows security>";
            var identity = user.Identity;
            return identity.IsAuthenticated
                ? identity.Name
                : "<Guest>";
        }
    }
}